server/src/main/java/chat/engine/Connect.java

```diff
@@ -0,0 +1,93 @@
+package chat.engine;
+
+import org.apache.log4j.Logger;
+
+import java.io.*;
+import java.net.Socket;
+import java.nio.charset.StandardCharsets;
+
+public class Connect implements Runnable {
+
+    private static final String CLIENT = "Client: ";
+
+    private static final String EXIT = "exit";
+
+    private static final Logger LOGGER = Logger.getLogger(Connect.class);
+
+    private String name = "";
+
+    private Socket socket;
+
+    private BufferedWriter writer;
+
+    private BufferedReader reader;
+
+    public Connect(Socket socket) throws IOException {
+        this.socket = socket;
+        writer = getWriter(socket);
+        reader = getReader(socket);
+    }
+
+    @Override
+    public void run() {
+
+        try {
+            writer.write("Please enter your name: ");
+            writer.flush();
+            name = reader.readLine();
+            LOGGER.info(CLIENT + name + " connected");
+
+        } catch (IOException e) {
+            LOGGER.info("context", e);
```

Comment:
what does it mean "context"?  Make some more informative message like "Failed to proceed message: "